### ISSUE TITLE
aws cloudwatch logs filter pattern include lines before and/or after matching pattern

### ISSUE THREAD

**Issue by HaithamMaya** [aws/aws-cli#5008]
Is there a way to include N lines before and/or after a matching pattern in AWS CloudWatch Logs?

Let's say I have this query and would like 3 lines before and after each match.

```
aws logs filter-log-events --log-group-name my-group --filter-pattern "mypattern"
```
The only work around I have at the moment is to remove the filter pattern and use grep:

```
aws logs filter-log-events --log-group-name my-group | grep -A 3 -B 3 mypattern
```

However, I would like to only stream the log events I need and do it as part of the aws log events query.

Ref:
https://stackoverflow.com/questions/59217130/aws-cloudwatch-logs-filter-pattern-include-lines-before-and-or-after-matching-pa

**Comment by KaibaLopez** [comment]
Hi @HaithamMaya ,
I don't think there is, so I'd say this a pretty good feature request.
I'll take a another look to make sure but yea, it would be a cool feature.

**Comment by felipepulpo** [comment]
This is necessary

**Comment by iloveitaly** [comment]
Agreed, it would *very* helpful to have this functionality built-in.

**Comment by shuangwu-fremtind** [comment]
Agreed + 1. Need it a lot.

**Comment by andyhush-athena** [comment]
I would love this functionality.

**Comment by diegonat** [comment]
+1

**Comment by jeremymcgee73** [comment]
+1 Want!

**Comment by JohnHillegass** [comment]
+1 as well. Definitely would be a huge win for Cloud Native log filtering!

**Comment by kdaily** [comment]
Thanks all for your feedback and enthusiasm around this issue!

This would be difficult to implement in the AWS CLI. There is one customization available for log filtering with the `aws logs tail` command. This is significantly more complex. We would need to implement our own filter pattern instead of relying on the existing one.

I've requested this feature with the CloudWatch logs team to see if it could be performed server-side and will update when I hear back.

**Comment by kdaily** [comment]
V345354868

**Comment by techish** [comment]
+1 !

**Comment by juliomfreitas** [comment]
+1

**Comment by rotherfuchs** [comment]
+1

**Comment by bharathsharma** [comment]
+1

**Comment by benwillkommen** [comment]
+1

**Comment by nileshprasad137** [comment]
+1

**Comment by omkar-asort** [comment]
+1

**Comment by kdaily** [comment]
The looging team is aware of the feature request. I don't have any other details on if or when it would be implemented on their side.

**Comment by sdurgi17** [comment]
+1 This would be a great addition. 
Not just in the cli, but also in the interface.

**Comment by k-1543** [comment]
+1

**Comment by rvong** [comment]
THIS IS NOT BAR RAISING CUSTOMER OBSESSION

**Comment by kdaily** [comment]
Hi all, thanks for the feedback. I'm checking in with the CloudWatch team to see if there are any other updates that I can provide on this issue, and will post here if I hear back.

Since this is an issue for the service team and not specific to the AWS CLI, I'm going to move this issue to the [AWS SDK repository](https://github.com/aws/aws-sdk/). The current link to this issue will redirect to the transferred repository.
